README.md

```diff
@@ -8,59 +9,66 @@
 [![Documentation Status](https://readthedocs.org/projects/pyteal/badge/?version=latest)](https://pyteal.readthedocs.io/en/latest/?badge=latest)
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 
-PyTeal is a Python language binding for [Algorand Smart Contracts (ASC1s)](https://developer.algorand.org/docs/features/asc1/). 
+PyTeal is a Python language binding for [Algorand Smart Contracts (ASC1s)](https://developer.algorand.org/docs/features/asc1/).
 
-Algorand Smart Contracts are implemented using a new language that is stack-based, 
-called [Transaction Execution Approval Language (TEAL)](https://developer.algorand.org/docs/features/asc1/teal/). 
+Algorand Smart Contracts are implemented using a new language that is stack-based,
+called [Transaction Execution Approval Language (TEAL)](https://developer.algorand.org/docs/features/asc1/teal/).
 
-However, TEAL is essentially an assembly language. With PyTeal, developers can express smart contract logic purely using Python. 
+However, TEAL is essentially an assembly language. With PyTeal, developers can express smart contract logic purely using Python.
 PyTeal provides high level, functional programming style abstractions over TEAL and does type checking at construction time.
 
-### Install 
+## Install
 
 PyTeal requires Python version >= 3.10.
 
 To manage multiple Python versions use tooling like [pyenv](https://github.com/pyenv/pyenv).
 
-#### Recommended: Install from PyPi
+### Recommended: Install from PyPi
 
 Install the latest official release from PyPi:
 
 * `pip install pyteal`
 
-#### Install Latest Commit
+### Install Latest Commit
 
 If needed, it's possible to install directly from the latest commit on master to use unreleased features:
 
 > **WARNING:** Unreleased code is experimental and may not be backwards compatible or function properly. Use extreme caution when installing PyTeal this way.
 
 * `pip install git+https://github.com/algorand/pyteal`
 
-### Documentation
+## Documentation
 
 * [PyTeal Docs](https://pyteal.readthedocs.io/)
 * `docs/` ([README](docs/README.md)) contains raw docs.
 
-### Development Setup
+## Development Setup
 
 Setup venv (one time):
- * `python3 -m venv venv`
+
+* `python3 -m venv venv`
 
 Active venv:
- * `. venv/bin/activate` (if your shell is bash/zsh)
- * `. venv/bin/activate.fish` (if your shell is fish)
 
-Pip install PyTeal in editable state
- * `pip install -e .`
+* `. venv/bin/activate` (if your shell is bash/zsh)
+* `. venv/bin/activate.fish` (if your shell is fish)
+
+Pip install PyTeal in editable state with dependencies:
+
+* `pip install -e.[development]`
```

Comment:
nit: should be `pip install -e . [development]`? I am not quite sure what `[development]` refers to.

Reply:
This PR removes `requirements.txt` which had several dependencies such as `black`, `mypy` and `pytest`. These have been moved into `setup.py` under `extras_require=`. Additionally, in another thread @jasonpaulos pointed out that it's unlikely anyone would want to `pip install` without the development requirements. So I removed that example and replaced with this. I'm definitely not wed to this position and am open to suggested improvements.

Reply:
I see what's going on my side, I am on `zsh`, and I have to use do `pip install -e .\[development\]`, nothing to worry about!

Reply:
Oh good catch. I'm using `zsh` too, but I've always just run `make setup-development` which works just fine. But you're right, when I tried the raw `pip` command I got the same error. I'll add a min-section to highlight this problem.

Reply:
A quick solution: https://kinopyo.com/en/blog/escape-square-bracket-by-default-in-zsh